sdk/Javascript/src/Platforms/Platform.FxOS.js

```diff
@@ -0,0 +1,277 @@
+    /// </param>
+
+    return getBestTransport().performRequest(request, callback);
+};
+
+exports.getUserAgent = function () {
+    // Browsers don't allow you to set a custom user-agent in ajax requests. Trying to do so
+    // will cause an exception. So we don't.
+    return null;
+};
+
+exports.getOperatingSystemInfo = function () {
+    return {
+        name: "--",
+        version: "--",
+        architecture: "--"
+    };
+};
+
+exports.getSdkInfo = function () {
+    var isCordovaEnvironment = window && window.cordova && window.cordova.version;
+
+    return {
+        language: isCordovaEnvironment ? "Cordova" : "Web",
+        fileVersion: $__fileVersion__
+    };
+};
+
+exports.login = function (startUri, endUri, callback) {
+    // Force logins to go over HTTPS because the runtime is hardcoded to redirect
+    // the server flow back to HTTPS, and we need the origin to match.
+    var findProtocol = /^[a-z]+:/,
+        requiredProtocol = 'https:';
+    startUri = startUri.replace(findProtocol, requiredProtocol);
+    endUri = endUri.replace(findProtocol, requiredProtocol);
+
+    return getBestProvider(knownLoginUis).login(startUri, endUri, callback);
+};
+
+exports.toJson = function (value) {
+    /// <summary>
+    /// Convert an object into JSON format.
+    /// </summary>
+    /// <param name="value" type="Object">The value to convert.</param>
+    /// <returns type="String">The value as JSON.</returns>
+
+    // We're wrapping this so we can hook the process and perform custom JSON
+    // conversions.  Note that we don't have to add a special hook to correctly
+    // serialize dates in ISO8061 because JSON.stringify does that by defualt.
+    // TODO: Convert geolocations once they're supported
+    // TODO: Expose the ability for developers to convert custom types
+    return JSON.stringify(value, function (key, stringifiedValue) {
+        if (dateSerializerOmitsDecimals && this && _.isDate(this[key])) {
+            // IE8 doesn't include the decimal part in its serialization of dates
+            // For consistency, we extract the non-decimal part from the string
+            // representation, and then append the expected decimal part.
+            var msec = this[key].getMilliseconds(),
+                msecString = String(msec + 1000).substring(1);
+            return stringifiedValue.replace(isoDateRegex, function (all, datetime) {
+                return datetime + "." + msecString + "Z";
+            });
+        } else {
+            return stringifiedValue;
+        }
+    });
+};
+
+exports.tryParseIsoDateString = function (text) {
+    /// <summary>
+    /// Try to parse an ISO date string.
+    /// </summary>
+    /// <param name="text" type="String">The text to parse.</param>
+    /// <returns type="Date">The parsed Date or null.</returns>
+
+    Validate.isString(text);
+
+    // Check against a lenient regex
+    var matchedDate = isoDateRegex.exec(text);
+    if (matchedDate) {
```

Comment:
Yes, it can. Firefox support ISO-8601 format.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse#ECMAScript_5_ISO-8601_format_support